src/watchdog_fsevents.c

```diff
@@ -717,14 +743,17 @@ watchdog_read_events(PyObject *self, PyObject *args)
 #endif
 
     /* Allocate information and store thread state. */
-    value = PyDict_GetItem(thread_to_run_loop, emitter_thread);
+    LOCK();
+    PyDict_GetItemRef(thread_to_run_loop, emitter_thread, &value);
+    UNLOCK();
```

Comment:
`PyDict_GetItemRef` is an atomic operation in CPython, so it'll be consistent with or without the locking.